### PR TITLE
Tab Deep link extended -c#

### PR DESCRIPTION
**Pull request by HunaidHanfee-MSFT**

### CODE REVIEW

File: samples/tab-deeplink/csharp/DeepLinkBot/Bots/DeepLinkBot.cs [hunk 1]
```diff
@@ -115,25 +116,44 @@ public Attachment AdaptiveDeepLinkCard(string userName, ITurnContext turnContext
...
-                                           {
-                                             Url=new Uri(tabUrlTask3),
-                                             Title = "Teams Apps"
-                                           }
+                                            new AdaptiveTextBlock(){Text="Teams Apps",Color=AdaptiveTextColor.Accent,Size=AdaptiveTextSize.Medium,HorizontalAlignment=AdaptiveHorizontalAlignment.Center,Spacing=AdaptiveSpacing.None}
+                                        },
+                                        SelectAction = new AdaptiveOpenUrlAction()
+                                        {
+                                            Url=new Uri(tabUrlTask3),
+                                            Title = "Teams Apps"
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Bots/DeepLinkBot.cs:133]: Check for spacing between = and other formatting.

File: samples/tab-deeplink/csharp/DeepLinkBot/Bots/DeepLinkBot.cs [hunk 2]
```diff
@@ -115,25 +116,44 @@ public Attachment AdaptiveDeepLinkCard(string userName, ITurnContext turnContext
...
+                                        Items=new List<AdaptiveElement>()
+                                        {
+                                            new AdaptiveTextBlock(){Text="Extended Deeplink features",Color=AdaptiveTextColor.Accent,Size=AdaptiveTextSize.Medium,HorizontalAlignment=AdaptiveHorizontalAlignment.Center,Spacing=AdaptiveSpacing.None}
+                                        },
+                                        SelectAction = new AdaptiveOpenUrlAction()
+                                        {
+                                            Url=new Uri(extendedDeepLink),
+                                            Title = "Extended Deeplink features"
                                         }
                                     }
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Bots/DeepLinkBot.cs:154]: Check for spacing between = and other formatting.

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+      margin: 1rem 8.5rem 1rem 11.5rem;
+  }
+
+  .form-container {
+      margin-left: 1rem;
+  }
+
+  .card-body {
+      margin-left: 0.5rem;
+  }
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:73]: We can move CSS in a separate file.(Use external CSS)

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+              attendees: attendeeInput.value,
+              content: contentInput.value.trim(),
+              endTime: endTimeInput.value,
+              startTime: startTimeInput.value,
+              subject: subjectInput.value.trim(),
+          };
+
+          //microsoftTeams.calendar.composeMeeting(meetingDetails).
+          //    then((result) => { console.log(result); }).
+          //    catch((error) => { console.log(error); });
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:129]: We can remove commented code.

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+
+          microsoftTeams.app.openLink(`https://teams.microsoft.com/l/meeting/new?subject=${meetingDetails.subject}&startTime=${meetingDetails.startTime}&endTime=${meetingDetails.endTime}&content=${meetingDetails.content}&attendees=${meetingDetails.attendees}`)
+      }
+  }
+
+  function onAppInstallDialogClick() {
+      var appInstallDetails = { appID: "c4c35992-6454-4b49-b8d3-b16ce2e36db2" };
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:136]: Can't we read APp ID from appsettings or pass it from Controller method?

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+      if(microsoftTeams.appInstallDialog.isSupported()) {
+          const dialogPromise = microsoftTeams.appInstallDialog.openAppInstallDialog(appInstallDetails).
+          then((result) => { console.log(result); }).
+            catch((error) => { console.log(error); });
+      }
+      else { console.log("App install capability not supported."); }
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:142]: App install capability is not supported.

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+              <pre>
+microsoftTeams.call.startCall({ targets: usersEmailArray,
+        requestedModalities: [microsoftTeams.call.CallModalities.Audio]})
+    .then((result) => {
+        console.log(result);
+    }).catch((error) => {
+            console.log(error);
+    });
+                           </pre
+              >
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:168]: Check for spacing and alignment

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+              <pre>
+microsoftTeams.call.startCall({ targets: usersEmailArray,
+    requestedModalities: [microsoftTeams.call.CallModalities.Video]})
+.then((result) => {
+    console.log(result);
+}).catch((error) => {
+        console.log(error);
+});
+						</pre
+              >
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:200]: Check for spacing and alignment

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+            attendees: emailArray,
+            content: "Meeting Content",
+            startTime: "2022-06-17T21:30+05:30",
+            endTime: "2022-06-17T22:30+05:30",
+            subject: "Meeting Subject",
+        })
+    .then((result) => { console.log(result); })
+    .catch((error) => { console.log(error); })
+                            </pre
+              >
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:233]: Check for spacing and alignment

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+          </div>
+          <button
+            data-id="singleSelect"
+            type="submit"
+            class="callDeeplinkButton"
+            style="width:130px;margin-left:8.5rem;"
+            onclick="onSchedulingDialogClick()"
+          >
+            Schedule Meeting
+          </button>
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:280]: Add proper line spacing

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Reviewed and provided my comments.

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml (hunk 4 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:129]: It is required there. Added another line after discussing with Wajeed - That this feature is not yet supported in Microsoft Teams.

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml (hunk 5 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:136]: This is suggested by Wajeed to put static polly app Id.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Did we updated the Readme file with new screenshots?

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving the PR.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**